### ISSUE TITLE
Nodejs Redis HGETALL return different values from two consecutive calls

### ISSUE THREAD

**Issue by CHYGO1985** [redis/node-redis#2427]
### Description

Hi there, 

I am having a weird issue here.

The scenario is like the following (all these happen in a local environment): I am using the node-redis, and I used `HSET` to store the following data: `{"value":"test3","isAvail":"true","refCount":"0","acctLevel":"0","username":"user3","id":"3"}`

Then I read the data from redis, and change refCount from '0' to '1', and save it back to redis via using  `HSET` again, and here is what happens using `HGETALL` to read from redis 3 times in a row, here is the console.log output:

`after1: {"value":"test3","isAvail":"true","refCount":"0","acctLevel":"0","username":"user3","id":"3"} after2: {"value":"test3","isAvail":"true","refCount":"1","acctLevel":"0","username":"user3","id":"3"} after3: {"value":"test3","isAvail":"true","refCount":"1","acctLevel":"0","username":"user3","id":"3"}`

I did nothing between the calls. My questions is why the first time it returns refCount as 0 and 2,3 time returns as '1' ?? Is there any cache mechanism in redis? (I tried the same thing via redis-cli: write -> read -> change refCount -> save back), and the first time I call hgetall it returns refCount as '1'

My redis version is : redis_version:6.2.5

### Node.js Version

v18.7.0

### Redis Server Version

redis_version:6.2.5

### Node Redis Version

redis@4.6.4

### Platform

macOS

**Comment by leibale** [comment]
@CHYGO1985 I tried to reproduce it using
```javascript
import { createClient } from 'redis';

const client = createClient();
client.on('error', err => console.error(err));

await client.connect();

console.log('HSET value 0', await client.hSet('key', { value: '0' }));
console.log('HGETALL', await client.hGetAll('key'));
console.log('HSET value 1', await client.hSet('key', { value: '1' }));
console.log('HGETALL', await client.hGetAll('key'));

await client.disconnect();
```
result
```
HSET value 0 1
HGETALL [Object: null prototype] { value: '0' }
HSET value 1 0
HGETALL [Object: null prototype] { value: '1' }
```

are you using replicas of any kind?

**Comment by CHYGO1985** [comment]
> @CHYGO1985 I tried to reproduce it using
> 
> ```js
> import { createClient } from 'redis';
> 
> const client = createClient();
> client.on('error', err => console.error(err));
> 
> await client.connect();
> 
> console.log('HSET value 0', await client.hSet('key', { value: '0' }));
> console.log('HGETALL', await client.hGetAll('key'));
> console.log('HSET value 1', await client.hSet('key', { value: '1' }));
> console.log('HGETALL', await client.hGetAll('key'));
> 
> await client.disconnect();
> ```
> 
> but this is what it printed
> 
> ```
> HSET value 0 1
> HGETALL [Object: null prototype] { value: '0' }
> HSET value 1 0
> HGETALL [Object: null prototype] { value: '1' }
> ```
> 
> are you using replicas of any kind?

thank you for your quick reply, let me double check my code and see if I can reproduce it in a simple sample

**Comment by leibale** [comment]
@CHYGO1985 did you manage to reproduce it?

**Comment by wmantly** [comment]
I can reproduce this pretty simply. Whenever I do `HGET` when there is nothing to be found, I get back `[Object: null prototype] {}` and it's very annoying.

**Comment by leibale** [comment]
@wmantly are you talking about `HGETALL`? because `HGET` never returns an object...
In redis, an empty hash and an empty key are equivalent, so `HGETALL` on an empty hash will return an empty object (you'll get the same results using `redis-cli`).

**Comment by wmantly** [comment]
@leibale yes, its `HGETALL`. I just updated from from an old version of this lib. In past versions something like this worked fine:
```js
let res = await client.HGETALL(`${this._name}_${data[this._key]}`);

// If the redis query resolved to something, prepare the data.
if(res){
   // do stuff with the data
}else{
  // throw error about res being empty
}
```

But now, since `HGETALL` returns `[Object: null prototype] {}`, and `[Object: null prototype] {}` is truthy, that code doesn't work,

**Comment by wmantly** [comment]
In past versions, im 99% sure `hgetall` returned `null` or `undefined` if it got nothing from redis. witch is far more helpful then an empty object.
